ibis/backends/flink/registry.py

```diff
@@ -34,6 +34,35 @@ def extract_field_formatter(translator: ExprTranslator, op: ops.Node) -> str:
     return extract_field_formatter
 
 
+def _cast(translator: ExprTranslator, op: ops.generic.Cast) -> str:
+    from ibis.expr.datatypes.core import Timestamp
+
+    arg, to = op.arg, op.to
+    if isinstance(to, Timestamp) and to.timezone:
+        arg_translated = translator.translate(arg)
+        return f"CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}')"
+
+    from ibis.backends.base.sql.registry.main import cast
+
+    return cast(translator=translator, op=op)
+
+
+def _interval_add(translator: ExprTranslator, op: ops.temporal.IntervalSubtract) -> str:
+    interval_left, interval_right = op.left, op.right
+    interval_left_translated = translator.translate(interval_left)
+    interval_right_translated = translator.translate(interval_right)
+    return f"{interval_left_translated} + {interval_right_translated}"
+
+
+def _interval_subtract(
+    translator: ExprTranslator, op: ops.temporal.IntervalSubtract
+) -> str:
+    interval_left, interval_right = op.left, op.right
+    interval_left_translated = translator.translate(interval_left)
+    interval_right_translated = translator.translate(interval_right)
+    return f"{interval_left_translated} - {interval_right_translated}"
```

Comment:
There's a lot of duplication for `_interval_add`, `_interval_subtract`, `_timestamp_add`, etc.; they could all basically be one function-creating function, that takes the operator as an argument (because you're just doing `return f"{translator.translate(op.left)} {op} {translator.translate(op.right)}"` for all of them.